### ISSUE TITLE
Multiple duplicate lines in choco console UI (but not log file)

### ISSUE THREAD

**Issue by opusforlife2** [chocolatey/ChocolateyGUI#933]
### What You Are Seeing?

When installing or updating a package, the console often shows multiple (2 or 3) duplicates of the log messages that show after the `Exporting function`, `Importing function`, and `Importing alias` lines.

One example:
```
[INFO ] - VERBOSE: Refreshing environment variables from the registry.
[INFO ] - VERBOSE: Refreshing environment variables from the registry.
[INFO ] - VERBOSE: Refreshing environment variables from the registry.
[INFO ] - VERBOSE: Refreshing environment variables from the registry.
[INFO ] - Restricting write permissions to Administrators
[INFO ] - Restricting write permissions to Administrators
[INFO ] - Restricting write permissions to Administrators
[INFO ] - We are setting up the Chocolatey package repository.
```

### What is Expected?

Each line should only show up once.

### How Did You Get This To Happen? (Steps to Reproduce)

1. Turn on "Show console output".
2. Install or update any package.
3. Notice the ongoing log, especially around the lines related to downloading and installing.

### System Details

* OS Build (In PowerShell run `[System.Environment]::OSVersion.version.tostring()`): Windows 7
* Windows PowerShell version (Run: `$PSVersionTable`): Chocolatey inbuilt
* Chocolatey version (Run `choco --version`): 

### Output Log

N/A - The log file correctly shows only one line for each message. This seems to be a UI problem only.

**Comment by JohnVeness** [comment]
I've been seeing this a lot in recent weeks. I'm using Windows 10, with Chocolatey 1.10 and ChocolateyGUI 1.0.0.0.

It worries me when I see things like the below, as it looks like it's downloading and installing twice:

```
Downloading libreoffice-fresh 64 bit
  from 'https://download.documentfoundation.org/libreoffice/stable/7.3.2/win/x86_64/LibreOffice_7.3.2_Win_x64.msi'
Downloading libreoffice-fresh 64 bit
  from 'https://download.documentfoundation.org/libreoffice/stable/7.3.2/win/x86_64/LibreOffice_7.3.2_Win_x64.msi'


Download of LibreOffice_7.3.2_Win_x64.msi (332.01 MB) completed.
Download of LibreOffice_7.3.2_Win_x64.msi (332.01 MB) completed.
Installing libreoffice-fresh...
Installing libreoffice-fresh...
```

**Comment by opusforlife2** [comment]
> It worries me when I see things like the below, as it looks like it's downloading and installing twice:

Indeed. That was the very reason my attention was grabbed by this bug. Downloading 600+ MB instead of 300+ MB would have been a huge waste of data. It was only after browsing the log that I realised this was not the case.

**Comment by ShowMeBillyJo** [comment]
I can reproduce this issue as well

![chocolatey-double-console-output](https://user-images.githubusercontent.com/179590/183114983-ad1b329b-9975-4713-b1cf-535564ad48cc.png)

**Comment by steviecoaster** [comment]
I hit this the other day while testing some things. 

![image](https://user-images.githubusercontent.com/8674240/184144057-29f183cd-296b-4549-9b44-41d47c00b9c2.jpeg)

@gep13 want me to do anything here to bring this issue up?

**Comment by DefectiveByDesign** [comment]
Has this been fixed yet? Here's some more info on this bug: The duplicate lines *increase* every time you install Chocolatey GUI. I got 3 lines after the latest update, uninstalled the GUI, reinstalled the GUI, now I have 4 lines duplicated. This perhaps isn't just a bug with the program, but possibly some resident config setting that doesn't get cleaned, and duplicates itself every time you install the GUI. I don't know where these settings are hidden, so I have no idea how to clean them myself.

**Comment by corbob** [comment]
@DefectiveByDesign unfortunately although we had hoped to get this fix into 1.1.0, the change I made in PR #954 introduced an even bigger and worse bug (in a certain scenario you lost **all** logging and all console output). However, knowing that the number of duplicate lines increases is good information. As for where the config is stored, we don't currently have it documented officially, but some of them are captured in a docs issue here: https://github.com/chocolatey/docs/issues/303

**Comment by ShowMeBillyJo** [comment]
@DefectiveByDesign You might be onto something. I'm 85–90% certain I've also un/reinstalled ChocolateyGUI in the past. It would've been a while back, though, hence the uncertainty.

**Comment by DefectiveByDesign** [comment]
It isn't just reinstalling, billyjo, simply updating to a new version will do it. As for the config files, is it safe to delete them? What would be the protocol for a clean uninstall / reinstall? I don't really understand the reasoning behind chocolatey not having a clean uninstall. This behavior is well known to be problematic with various software. Drivers for instance, created a need for DDU, and there are various uninstall tools, which some charge money for, that will monitor software installation to do "true" uninstallation. Some other programs I've known to need special cleaning has been various DRM, virtual CD software with drivers, motherboard utilities like Asus, Dell bloatware. There used to be stuff like PC decrapifier, which I think is no longer supported, but the need still exists. I shouldn't have to wipe the OS to fix software that doesn't uninstall properly.

Recently, updates in general seem to be a problem with chocolately, more than years ago. Many programs refuse to update. I have to uninstall them and reinstall, instead of using the update function. Sometimes the update function breaks the program even if it installs, like LibreOffice, which would load the writer then crash, which I fixed by uninstalling / reinstall. These problems don't exist with the old fashioned update process, just when using chocolatey, and it isn't with all updates either, so maybe the majority updates, but you're always left with a couple that don't. Kinda defeats the purpose.

**Comment by JohnVeness** [comment]
> The duplicate lines _increase_ every time you install Chocolatey GUI.

I'm not denying whether or not the above is true, but for interest I have just installed Chocolatey and Chocolatey-GUI onto a brand-new Windows 11 laptop, used Chocolatey-GUI to install Firefox and saw duplicate lines (2 lines) immediately.

**Comment by 274188A** [comment]
> > The duplicate lines _increase_ every time you install Chocolatey GUI.
> 
> I'm not denying whether or not the above is true, but for interest I have just installed Chocolatey and Chocolatey-GUI onto a brand-new Windows 11 laptop, used Chocolatey-GUI to install Firefox and saw duplicate lines (2 lines) immediately.


This has been going on for years - duplicate output lines....

![image](https://github.com/chocolatey/ChocolateyGUI/assets/40014477/8b38b862-0af5-4b8b-a091-6ff3524bb047)

**Comment by oasesoftware** [comment]
I'm rugging on chocolatey 2.2.2 and chocolateygui 2.1.0 and also have this multiple line output issue.
Is there a time-frame when this bug is solved, or for the first response, are only the log messages multyplied or are even the downloads and install proccesses multyplied?

**Comment by danrhodes** [comment]
The issue still exists in v3.0.0.0 I need to set time aside to look at this!

**Comment by danrhodes** [comment]
Turned out to be such an easy fix! #1129

**Comment by danrhodes** [comment]
@corbob in a previous pr you closed you were on the rights tracks but I fail to see why this fix caused what you said??

"We can't just tell it not to append the logger. If we do that, we cause a regression of #697. But we can use the instance we already have, and just backup and restore the config."

I will have a closer look later though.
